pkg/controller/controller_test.go

```diff
@@ -96,9 +75,3 @@ func TestGetSelfDeploy(t *testing.T) {
 		require.Nil(t, self)
 	})
 }
-
-func TestNewManagerForRestConfig(t *testing.T) {
```

Comment:
can't test things that use the utils.upsert function with the fake environment. That's fine. NewManagerForRestConfig is essentially the main function and doesn't have any logic by itself. Isn't valuable to unit test in the first place. 

This will be covered by all e2e tests as well.